src/reflect/value.go

```diff
@@ -1061,8 +1083,8 @@ func (it *MapIter) Next() bool {
 func (v Value) Set(x Value) {
 	v.checkAddressable()
 	v.checkRO()
-	if !x.typecode.AssignableTo(v.typecode) {
-		panic("reflect: cannot set")
+	if !x.typecode.AssignableTo(v.typecode) && !x.typecode.ConvertibleTo(v.typecode) {
```

Comment:
Should this be `||` instead of `&&` ?

Reply:
According to the go docs:

> As in Go, x's value must be assignable to v's type and must not be derived from an unexported field.

So this change shouldn't be needed... However, I was not getting correct behavior matching Go without it. 

I'll see if I can add some tests and update AssignableTo with whatever implicit conversions are allowed, instead. 

But yes, the && was correct because it was not A and not B, meaning not (A or B), i.e fail if not assignable or convertible.

Reply:
Thanks.  I think I must've missed one of the `!`'s when I skimmed this.